Source/Core/Core/Boot/DolReader.cpp

```diff
@@ -124,3 +140,80 @@ bool DolReader::LoadIntoMemory(bool only_in_mem1) const
 
   return true;
 }
+
+// On a real console this would be done in the Espresso bootrom
+bool DolReader::LoadAncastIntoMemory() const
+{
+  // The ancast image will always be in data section 0
+  const auto& section = m_data_sections[0];
+  const u32 section_address = m_dolheader.dataAddress[0];
+
+  const EspressoAncastHeader* header = (const EspressoAncastHeader*)section.data();
+
+  // Verify header block size
+  if (Common::swap32(header->header_block.header_block_size) != sizeof(AncastHeaderBlock))
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid header block size: 0x{:x}", Common::swap32(header->header_block.header_block_size));
+    return false;
+  }
+
+  // Make sure this is a PPC ancast image
+  if (Common::swap32(header->signature_block.signature_type) != 0x01)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid signature type: 0x{:x}", Common::swap32(header->signature_block.signature_type));
+    return false;
+  }
+
+  // Make sure this is a Wii-Mode ancast image
+  if (Common::swap32(header->info_block.image_type) != ANCAST_IMAGE_TYPE_ESPRESSO_WII)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid image type: 0x{:x}", Common::swap32(header->info_block.image_type));
+    return false;
+  }
+
+  // Verify the body size
+  const u32 body_size = Common::swap32(header->info_block.body_size);
+  if (body_size + sizeof(EspressoAncastHeader) > section.size())
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid body size: 0x{:x}", body_size);
+    return false; 
+  }
+
+  // Verify the body hash
+  const auto digest = Common::SHA1::CalculateDigest(section.data() + sizeof(EspressoAncastHeader), body_size);
+  if (std::memcmp(digest.data(), header->info_block.body_hash, Common::SHA1::DIGEST_LEN) != 0)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Body hash mismatch");
+    return false; 
+  }
+
+  // Check if this is a retail or dev image
+  bool is_dev = false;
+  if (Common::swap32(header->info_block.console_type) == ANCAST_CONSOLE_TYPE_DEV) 
+    is_dev = true;
+  else if (Common::swap32(header->info_block.console_type) != ANCAST_CONSOLE_TYPE_RETAIL)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid console type: 0x{:x}", Common::swap32(header->info_block.console_type));
+    return false; 
+  }
+
+  // Decrypt the Ancast image
+  constexpr u8 vwii_ancast_retail_key[0x10] = { 0x2e, 0xfe, 0x8a, 0xbc, 0xed, 0xbb, 0x7b, 0xaa, 0xe3, 0xc0, 0xed, 0x92, 0xfa, 0x29, 0xf8, 0x66 };
+  constexpr u8 vwii_ancast_dev_key[0x10] = { 0x26, 0xaf, 0xf4, 0xbb, 0xac, 0x88, 0xbb, 0x76, 0x9d, 0xfc, 0x54, 0xdd, 0x56, 0xd8, 0xef, 0xbd };
+  std::unique_ptr<Common::AES::Context> ctx = Common::AES::CreateContextDecrypt(is_dev ? vwii_ancast_dev_key : vwii_ancast_retail_key);
+
+  std::vector<u8> decrypted;
```

Comment:
```suggestion
  std::vector<u8> decrypted(body_size);
```
...to allocate a vector with the correct size from the get-go, and then you can get rid of the .resize()